lib/Common/Memory/Recycler.cpp

```diff
@@ -4629,7 +4673,19 @@ Recycler::IsConcurrentSweepSetupState() const
 BOOL
 Recycler::IsConcurrentSweepState() const
 {
-    return this->collectionState == CollectionStateConcurrentSweep;
+#if ENABLE_CONCURRENT_GC && ENABLE_ALLOCATIONS_DURING_CONCURRENT_SWEEP
+    if (CONFIG_FLAG_RELEASE(EnableConcurrentSweepAlloc))
+    {
+        return this->collectionState == CollectionStateConcurrentSweepPass1 ||
+            this->collectionState == CollectionStateConcurrentSweepPass1Wait ||
```

Comment:
Interesting- before, we'd be in ConcurrentSweepState if there was work going on in the background thread- now, this could be true even if we're in the foreground thread?

Reply:
Yes, that's right. I think it should be OK because the two pass sweep does do processing in foreground thread as well.

---
In reply to: [148170831](https://github.com/Microsoft/ChakraCore/pull/4104#discussion_r148170831) [](ancestors = 148170831)